A collection of bug fixes

Fesaa opened the pull request:
# Changed
- Changed: Reading lists will now only show one of promote & unpromote
- Changed: Collections will now only show one of promote & unpromote
- Changed: The reset password form will now be disabled if you do not have permissions to change your password (Fixes #3818)
- Changed: The ToC in the EPUB reader will now mark a chapter as current until the next chapter is reached (Fixes #3817)
- Changed: Go to Page prompts on book/manga reader now use a custom styled window
- Changed: Removed the opacity from tooltips

# Fixed
- Fixed: Fixed being unable to disable/enable automatic webtoon detection (Fixes #3793)
- Fixed: Fixed the volume modal missing the word count header (Fixes #3806)
- Fixed: Fixed search shortcut hint appearing on small devices (Fixes #3810) (develop)
- Fixed: Fixed actionables not showing up correctly on mobile (Fixes #3808) (develop)
- Fixed: Fixed preferences not displaying until locale took too long to load in (Fixes #3789)
- Fixed: Fixed the section jumper in the EPUB reader using the wrong terms (Fixes #3819)
- Fixed: Fixed download actionable missing for Admins in some cases 
- Fixed: Fixed missing download handlers on series detail for volume/chapter entities. (Fixes #3830)

Fesaa committed:
Collection of smaller bugs
#3793, #3806, #3810, #3808

Fesaa committed:
Fix potential race condition with locale, most likely fixes #3789

Fesaa committed:
Disable change password form if the user can't change their password Fixes #3818

Fesaa committed:
Use correct naming in the section jumper Fixes #3819

Fesaa committed:
Mark chapter as selected as long as the next chapter is not reached Fixes #3817

Also adds a colour to the selected chapter, to make it more clear

Fesaa committed:
Same promote/unpromote logic but for collections

Found a bug where the shouldRender isn't called again when the entity
state updates!

majora2007 commented on UI/Web/src/app/collections/_components/collection-detail/collection-detail.component.ts:
```diff
@@ -225,6 +226,17 @@ export class CollectionDetailComponent implements OnInit, AfterContentChecked {
     });
   }
 
+  shouldRenderCollection(action: ActionItem<UserCollection>, entity: UserCollection, user: User) {
```
These should be named shouldRenderCollectionAction to align with other code signatures

majora2007 commented on UI/Web/src/assets/langs/en.json:
```diff
@@ -882,7 +882,7 @@
         "next": "Next",
         "previous": "Previous",
-        "go-to-page-prompt": "There are {{totalPages}} pages. What page do you want to go to?"
+        "go-to-section-prompt": "There are {{totalSections}} sections. What section do you want to go to?"
```
The code around pages/sections doesn't make sense to me. An epub has pages (the physical pages) and sections for when the reader breaks it into column mode.

Fesaa replied:
Yes, the naming was also a little confusing to me. I might want to leave an extra comment on the method. 

The changes were for this issue: https://github.com/Kareadita/Kavita/issues/3819; if that clears it up for you?

majora2007 committed:
Revert "Use correct naming in the section jumper Fixes #3819"

This reverts commit db0f1dae8604ade89553846e89fe3b399208fb77.

majora2007 committed:
Fixed tooltips being slightly opaque.

Refactored Amelia's attempt at clearing up the section code and added some new controls to make paging easier.

Book/Manga reader go to page is now using a custom styled prompt instead of the basic window.prompt.

This code contains goToSection() that is not implemented.

Fesaa committed:
Revert, and only keep the off by one fix & new prompt

Fesaa committed:
Fix alignment on collection details page

Fix overlap with manga reader in go-to-page dialog

majora2007 committed:
Fixed download actionable missing in some cases due to a failed admin check.

Fixed download missing on series-detail page.

majora2007 committed:
API documentation
